src/tools/mesh_tester.cpp

```diff
@@ -250,6 +269,124 @@ std::vector< std::pair<int, int> > naiveIntersectionAlgorithm(
   return retval;
 }
 
+template < typename ExecSpace >
+std::vector< std::pair<int, int> > naiveIntersectionAlgorithm(
+  mint::Mesh* surface_mesh,
+  std::vector<int> & degenerate)
+{
+  SLIC_INFO("Execution Space is " <<
+    axom::execution_space< ExecSpace >::name());
+
+  // Get allocator
+  int allocatorID = axom::execution_space< ExecSpace >::allocatorID();
+  umpire::Allocator allocator = axom::getAllocator(allocatorID);
+
+  std::vector< std::pair<int, int> > retval;
+
+  const int ncells = surface_mesh->getNumberOfCells();
+  SLIC_INFO("Checking mesh with a total of "<< ncells<< " cells.");
+
+  Triangle3 * tris = axom::allocate <Triangle3> (ncells, allocator);
+  
+  // Get each triangle in the mesh
+  for (int i = 0; i < ncells ; i++)
+  {
+    Triangle3 t1 = getMeshTriangle(i, surface_mesh);
+    tris[i] = t1;
+    if (t1.degenerate())
+    {
+      degenerate.push_back(i);
+    }
+  }
+
+  RAJA::RangeSegment row_range(0, ncells);
+  RAJA::RangeSegment col_range(0, ncells);
+
+  // Kernel policy
```

Comment:
@gzagaris -- is there a cleaner way to define these (nested) kernels using the execution policies that you set up?

Reply:
Based on conversation with @rhornung67 , we opted not to include convenience aliases for RAJA::Kernel since it can be too restrictive and lead to less performant implementations, which would allow users to use RAJA in a way they are not supposed to. Instead we push that to the application. 

We should revisit this and think about this more. If nothing else, we should provide the means to define those using our predefined axom::* policies